Automattic/wp-calypso
# Reader: Convert SiteStream to functional component and add sidebar

## Proposed Changes

* Converts SiteStream to a functional component
* Adds sidebar with post count, follower count, and most used post tags
* Moves common components and functions for access by both FeedStream and SiteStream

This PR replicates the changes from https://github.com/Automattic/wp-calypso/pull/78702 and https://github.com/Automattic/wp-calypso/pull/78706 done for the FeedStream component.

## Testing Instructions

* Visit `/reads/blogs/{blog-id}` to see the sidebar added.

| **Before** | **After** |
| - | - |
| <img width="733" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1699996/14b95881-3b36-440e-a89e-7acde6bd2bac"> | <img width="945" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1699996/f22bf5de-858f-4ac9-9484-d2bdf3884b96"> |
